core/src/main/java/org/kiwix/kiwixmobile/core/utils/BookUtils.kt

```diff
@@ -15,46 +15,44 @@
  * along with this program. If not, see <http://www.gnu.org/licenses/>.
  *
  */
-package org.kiwix.kiwixmobile.core.utils;
+package org.kiwix.kiwixmobile.core.utils
 
-import java.util.HashMap;
-import java.util.Locale;
-import java.util.Map;
+import java.util.HashMap
+import java.util.Locale
 
 /**
  * Created by mhutti1 on 19/04/17.
  */
-
-public class BookUtils {
-
-  public final Map<String, Locale> localeMap;
+class BookUtils {
+  val localeMap: Map<String, Locale>
 
   // Create a map of ISO 369-2 language codes
-  public BookUtils() {
-    String[] languages = Locale.getISOLanguages();
-    localeMap = new HashMap<>(languages.length);
-    for (String language : languages) {
-      Locale locale = new Locale(language);
-      localeMap.put(locale.getISO3Language(), locale);
-    }
+  init {
+    val languages = Locale.getISOLanguages()
+    localeMap = HashMap(languages.size)
+    languages
+      .asSequence()
+      .map(::Locale)
+      .forEach { localeMap.put(it.isO3Language, it) }
   }
 
   // Get the language from the language codes of the parsed xml stream
-  public String getLanguage(String languageCode) {
-
+  fun getLanguage(languageCode: String?): String {
+    var language = ""
     if (languageCode == null) {
-      return "";
-    }
-
-    if (languageCode.length() == 2) {
-      return new LanguageContainer(languageCode).getLanguageName();
-    } else if (languageCode.length() == 3) {
-      try {
-        return localeMap.get(languageCode).getDisplayLanguage();
-      } catch (Exception e) {
-        return "";
+      language = ""
+    } else {
+      if (languageCode.length == 2)
+        language = LanguageContainer(languageCode).languageName
+      else if (languageCode.length == LANGUAGE_CODE_LENGTH_THREE) {
+        val locale = localeMap[languageCode]
+        language = locale?.displayLanguage.toString()
       }
     }
-    return "";
+    return language
+  }
+
+  companion object {
+    const val LANGUAGE_CODE_LENGTH_THREE = 3
```

Comment:
There isn't really a reason to extract a constant for the number three, we can't give it any more meaning than "its three"

Reply:
@macgills Should I ignore if "staticAnalysis" fails while pushing?
`
Logs:
> Configure project :app
[directory 'C:\Users\Hritik\StudioProjects\kiwix-android\app\build', directory 'C:\Users\Hritik\StudioProjects\kiwix-android\core\build', directory 'C:\Users\Hritik\StudioProjects\kiwix-android\custom\build']
> Task :core:detekt
Ruleset: style - 10min debt
	MagicNumber - [getLanguage] at C:\Users\Hritik\StudioProjects\kiwix-android\core\src\main\java\org\kiwix\kiwixmobile\core\utils\BookUtils.kt:47:39

Overall debt: 10min

Build failed with 1 weighted issue (threshold defined was 0).

> Task :core:detekt FAILED
1`

Reply:
We have a couple of options, we can [suppress it](https://arturbosch.github.io/detekt/suppressing-rules.html) in this instance or add the number 3 as an exception in the detekt rule in detekt.yml
```
MagicNumber:
    active: true
    excludes: "**/test/**,**/androidTest/**,**/*.Test.kt,**/*.Spec.kt,**/*.Spek.kt"
    ignoreNumbers: '-1,0,1,2'
    ignoreHashCodeFunction: true
    ignorePropertyDeclaration: false
    ignoreLocalVariableDeclaration: false
    ignoreConstantDeclaration: true
    ignoreCompanionObjectPropertyDeclaration: true
    ignoreAnnotation: false
    ignoreNamedArgument: true
    ignoreEnums: false
    ignoreRanges: false
```

I think just suppress it for this class because 3 could very well be a magic number elsewhere but if just feels unnecessary here.